src/libmongoc/tests/json-test-monitoring.c

```diff
@@ -442,6 +442,25 @@ _apm_match_error_context (const bson_t *actual, const bson_t *expectations)
    bson_free (actual_str);
    bson_free (expectations_str);
 }
+
+bool skip_cse_list_collections(const bson_t *doc) {
+   // see CDRIVER-3856: Sharing a MongoClient for metadata lookup can lead to
+   // deadlock in drivers using automatic encryption
+   const char* val;
+
+   if (!bson_has_field(doc, "command_started_event.command.listCollections"))
+      return false;
```

Comment:
If this returns true if we should skip the check, I would think this should be:

```
if (bson_has_field(doc, "command_started_event.command.listCollections"))
    return true;

if (bson_has_field(doc, "command_started_event.command.$db") && 0 == strcmp (bson_lookup_utf8(doc, "command_started_event.command.$db"), "keyvault"))
    return true;

return false;

Reply:
I thought the logic here was to skip _only_ the `listCollections` commands on the "keyvault" database. As you wrote it, this function would skip all `listCollections` commands _and_ any commands on the "keyvault" database. I updated this function to do that but just wanted to double-check that was the intent.